Add entitlement related to notifyd blocking to WebContent Development

pvollan opened the pull request:
#### 693ae98c2146edd20bf2bb087eda06c9dc1ce48f
<pre>
Add entitlement related to notifyd blocking to WebContent Development
<a href="https://bugs.webkit.org/show_bug.cgi?id=274265">https://bugs.webkit.org/show_bug.cgi?id=274265</a>
<a href="https://rdar.apple.com/128139234">rdar://128139234</a>

Reviewed by Sihui Liu.

Add entitlement related to notifyd blocking to the WebContent Development XPC service, since it is also required there.
Without this entitlement, we can get into a notification loop state where notifications are continously being reposted
by the WebContent process. With this entitlement, notifyd will be blocked in the sandbox, preventing this loop.

* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/278899@main">https://commits.webkit.org/278899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/15166573f0ad21a37bc85c668e081ab7c7a6b138

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51844 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31156 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4208 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55112 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2536 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37530 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2244 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42170 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53943 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44718 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23303 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/51692 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26054 "Passed tests") | | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48010 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2070 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56703 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26965 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1953 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49573 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28202 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44815 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48823 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-playing-audio (failure)") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/11331 "The change is no longer eligible for processing.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29099 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7583 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27939 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (approved):
Can you update commit message about why the entitlement is needed?

aproskuryakov commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -198,37 +198,40 @@ function webcontent_sandbox_entitlements()
-    plistbuddy Add :com.apple.private.darwin-notification.introspect:18 string com.apple.system.timezone
-    plistbuddy Add :com.apple.private.darwin-notification.introspect:19 string com.apple.zoomwindow
-    if [[ "${WK_PLATFORM_NAME}" == macosx ]]
+    if [[ "${WK_USE_RESTRICTED_ENTITLEMENTS}" == YES ]]
```
Does WebKit still work in open source builds? This is more of a question about what was done before, not about this PR, but this is the first time I noticed.

pvollan replied:
That is a good point, Alexey. I believe so, since this entitlement is not private.

Thanks for reviewing!

aproskuryakov replied:
Surprised that it's not private when its name starts with `com.apple.private`, but more importantly, we aren't adding it in open source builds, correct? Does WebKit work without the entitlement?

pvollan replied:
That's a good point, Alexey. This patch is actually adding the entitlement `com.apple.developer.web-browser-engine.restrict.notifyd`, also to open source builds. The entitlement `com.apple.private.darwin-notification.introspect` is not being added to new configurations in this patch.

pvollan committed:
Add entitlement related to notifyd blocking to WebContent Development
https://bugs.webkit.org/show_bug.cgi?id=274265
rdar://128139234

Reviewed by Sihui Liu.

Add entitlement related to notifyd blocking to the WebContent Development XPC service, since it is also required there.
Without this entitlement, we can get into a notification loop state where notifications are continously being reposted
by the WebContent process. With this entitlement, notifyd will be blocked in the sandbox, preventing this loop.

* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: https://commits.webkit.org/278899@main
